weather/api_data.py

```diff
@@ -0,0 +1,30 @@
+from abstract_ds import Service
+import requests
+import matplotlib.pyplot as plt
+
+
+class APICallingService(Service):
+    """API data class for fetching and plotting temperature data."""
+
+    def get_data(self) -> dict:
+        """This method gets data from API for New York and returns it as a dictionary."""
+        response = requests.get('https://api.open-meteo.com/v1/forecast?latitude=40.7143&longitude=-74.006&current=temperature_2m&hourly=temperature_2m&temperature_unit=fahrenheit&wind_speed_unit=mph&precipitation_unit=inch')
```

Comment:
Instead of hardcoding the parameters into the the URL directly here, I would rather take these as input of the `get_data` method, and use those variables for a more flexible implementation of this.

An example of passing parameters using variables for your HTTP request:
```
        base_url = "https://api.open-meteo.com/v1/forecast"
        params = {}
        params['latitude'] = latitude    # float
        params['longitude'] = longitude    # float
        params['hourly'] = daily_weather_variable    # string with "temperature_2m"

        response = requests.get(base_url, params=params)
```